### PR TITLE
core(domstats): support an empty html body

### LINKED ISSUE #8725: Dom Size fails when no body present
This issue #6806 brought up counting elements in the HEAD, and this PR was made #7241.  But it fails to count dom elements in a website like this:

```html
<html>Content</html>
```

Where there are no `<body>` tags present, but there are nodes to count.

### PR DESCRIPTION
**Pull request by brendankenny**
fixes #8725

The issue was that domstats starts with a 0 as the max number of children seen as the base case, an empty `<body>` has 0 children (which isn't more than 0), and so `parentWithMostChildren` stayed as `null`.

Instead we start with a -1 so that there will always be at least `<body>` in there. Could have also gone with a `>=` check, but that changes it so `parentWithMostChildren` is the last one seen if there are ties, which would sort of be a breaking change. This seemed better (and more the original intent, I guess).

### CODE REVIEW

File: lighthouse-core/gather/gatherers/dobetterweb/domstats.js [hunk 1]
```diff
@@ -86,8 +86,8 @@ function elementPathInDOM(element) {
 /* istanbul ignore next */
 function getDOMStats(element, deep = true) {
   let deepestElement = null;
-  let maxDepth = 0;
-  let maxWidth = 0;
+  let maxDepth = -1;
```
Comment (brendankenny) [inline lighthouse-core/gather/gatherers/dobetterweb/domstats.js]: `maxDepth` doesn't need to change (`<body>` counts as depth 1 which is > 0), but it seemed weird not to change it as well :)

**Comment by brendankenny** [comment]
> nice, should we get a smoketest in there?

It is unfortunate to not have a regression test, but it felt like overkill to add a smoke test (+5 seconds to all future test runs?) for a trivial corner case of a single audit that can only occur with completely empty pages...

Where do you fall on that trade off?
